reference/pipeline/expressions/index.md

```diff
@@ -0,0 +1,230 @@
+---
+layout: single
+title:  "Pipeline Expression Reference"
+sidebar:
+  nav: reference
+---
+
+{% include toc %}
+
+Pipeline expressions allow you to reference arbitrary values about the
+state of your system in the execution of your pipelines. For more information
+about how pipeline expressions work, see the
+[overview](/guides/user/pipeline/expressions/).
+
+## Syntax elements
+
+Each pipeline expression is made of `$` followed by opening/closing brackets:
+`${ }`. Within these brackets, you can add arbitrary expressions to access and
+modify the details of your pipeline during pipeline execution.
+
+Note that expressions can't be nested: `${ expression1 ${expression2} }` won't
+be evaluated.
+
+### Code
+
+Spinnaker allows you to execute Java code within a pipeline expression. This can
+be useful for string manipulation or to build more complex expressions than
+would otherwise be possible. For security reasons, you can only use whitelisted
+Java classes. You can find the full list of whitelisted classes
+[here](#whitelisted-java-classes).
+
+You can use methods available to the whitelisted classes without importing them.
```

Comment:
this line doesn't really make sense. There isn't a syntax for importing a class in the expression language. I would eliminate it because it is confusing

Reply:
`You can use methods directly on existing map values based on their Java class`, is probably a better way of saying this